hs_modelinstance/static/js/model_instance.js

```diff
@@ -1,16 +1,108 @@
 
-// subscribe to the OnChange event for the model_name listbox
-var model_list = document.getElementById('id_model_name');
-if(model_list){
-    model_list.addEventListener('change', show_model_details);
+// variable to store the last submitted model program id
+var mp_old_id = '';
+
+/*
+  Initialize the view on document ready
+ */
+$(document).ready(function(){
+
+    // proceed only if selectbox_ exists (i.e. edit page)
+    if($('[id^=selectbox_]').length > 0) {
+
+        // set the initial value of the mp_old_id
+        mp_old_id = $('[id^=selectbox_]').attr('id').split('_').pop();
+
+        // initialize the model program listbox
+        build_selectbox();
+
+        // set the initial selection
+        set_model_program_selection();
+
+        // build metadata table (ajax query)
+        show_model_details();
+
+
+    }
+
+});
+
+/*
+  Sets the selected item in the selectbox if a ModelProgram foreign key relationship has previously been set in the
+  database.
+ */
+function set_model_program_selection(){
+
+        // get the value stored in the model_name hidden field
+        var mp_id = $('[id^=selectbox_]').attr('id').split('_').pop();
+
+        // set the initial selection
+        var model_name = document.getElementById('id_model_name');
+        model_name.value = mp_id;
+
+        // get the selectbox
+        var options = $('[id^=selectbox_]').find('option');
+
+        if (mp_id != "")
+        {
+            for (var i = 0; i< options.length; i++){
+                var o = options[i];
+                if (o.value == mp_id) {
+                    $(o).attr("selected", 1);
+                    break;
+                }
+            }
+        }
+        else {
+            // set selected to 'Not Specified' if mp_id is empty
+            $(options[0]).attr("selected", 1);
+        }
+}
+
+
+/*
+ Constructs a select box for the ModelProgram elements using JQuery.  This selectbox is used to define the
+ ModelProgram foreign key relationship.
+ */
+function build_selectbox(){
+
+    // destroy any existing multiselect widgets
+    $('.div-selectbox .selectbox').multiselect('destroy');
```

Comment:
The best approach for this would be to assign an ID to the selectbox you want to destroy, or include parent's ID (that is not expected to contain more 'selectbox' items) in the selector. 

This code needs to avoid destroying other elements using the class 'selectbox'. Might not be a problem right now, but could be a problem in the future when the class gets reused.